cmd/cstor-volume-mgmt/controller/volume-controller/handler.go

```diff
@@ -316,6 +386,166 @@ func (c *CStorVolumeController) getVolumeResource(
 	return cStorVolumeGot, nil
 }
 
+// GetCVStatusCapacity returns the capacity in form of resourceList
+// TODO: Move below code to cstorvolumepackage
+func GetCVStatusCapacity(cvObj *apis.CStorVolume) (corev1.ResourceList, error) {
+	resCapacity, err := resource.ParseQuantity(cvObj.Spec.Capacity)
+	if err != nil {
+		return nil, pkg_errors.Wrapf(
+			err, "failed to update capacity: failed to parse capacity {%s}",
+			cvObj.Spec.Capacity,
+		)
+	}
+	resourceList := corev1.ResourceList{
+		corev1.ResourceName(corev1.ResourceStorage): resCapacity,
+	}
+	return resourceList, nil
+}
+
+// resizeCStorVolume resize the cstorvolume and if any error occurs updates the
+// resize conditions of cstorvolume either with success or failure message
+func (c *CStorVolumeController) resizeCStorVolume(cStorVolume *apis.CStorVolume) error {
+	var eventMessage string
+	var err error
+	cloneCV := cStorVolume.DeepCopy()
+	customCVObj := cstorvolume.NewForAPIObject(cloneCV)
+	conditionStatus := customCVObj.GetCVCondition(apis.CStorVolumeResizing)
+
+	err = c.resizeVolume(cloneCV)
+	if err != nil {
+		eventMessage = fmt.Sprintf(
+			"failed to resize cstorvolume %s with capacity %s error %v",
+			cloneCV.Name,
+			cloneCV.Spec.Capacity,
+			err,
+		)
+		c.recorder.Event(cloneCV, corev1.EventTypeWarning, string(common.FailureUpdate), eventMessage)
```

Comment:
Since we need to update the CV resize conditions with error message not to `swallow` the original error message We generated the events anyhow removed  event generation in caller code.